test/cppsuite/tests/cache_resize.cpp

```diff
@@ -0,0 +1,202 @@
+                 * dummy data.
+                 */
+                const uint64_t collection_id = 0;
+                const std::string key = "";
+                const std::string value = "";
+                tc->transaction.begin();
+                testutil_check(tc->tracking->save_operation(tc->session, tracking_operation::CUSTOM,
+                  collection_id, key, value, tc->tsm->get_next_ts(), tc->op_track_cursor));
+                testutil_assert(tc->transaction.commit());
+            }
+            increase_cache = !increase_cache;
+        }
+    }
+
+    void
+    insert_operation(test_harness::thread_context *tc) override final
+    {
+        uint64_t coll_id = 0;
+        collection &coll = tc->db.get_collection(coll_id);
+        scoped_cursor cursor = tc->session.open_scoped_cursor(coll.name);
+
+        while (tc->running()) {
+
+            tc->transaction.try_begin();
+
+            /* Insert a random value. */
+            const std::string value =
+              random_generator::instance().generate_pseudo_random_string(tc->value_size);
+            bool ret = tc->insert(cursor, coll.id, value);
+
+            if (!ret) {
+                tc->transaction.rollback();
+            } else if (tc->transaction.can_commit()) {
+                /*
+                 * The transaction can fit in the current cache size and is ready to be committed.
+                 * This means the tracking table will contain a new record to represent this
+                 * transaction which will be used during the validation stage.
+                 */
+                testutil_assert(tc->transaction.commit());
+            }
+
+            tc->sleep();
+        }
+
+        /* Make sure the last transaction is rolled back now the work is finished. */
+        if (tc->transaction.active())
+            tc->transaction.rollback();
+    }
+
+    void
+    validate(const std::string &operation_table_name, const std::string &,
+      const std::vector<uint64_t> &) override final
+    {
+        WT_DECL_RET;
+        int tracked_op_type;
+        tracking_operation op_type;
+        uint64_t num_records = 0;
+        wt_timestamp_t prev_ts = 0, tracked_ts;
+
+        /* Open a cursor on the tracking table to read it. */
+        scoped_session session = connection_manager::instance().create_session();
+        scoped_cursor cursor = session.open_scoped_cursor(operation_table_name);
+
+        /* Parse the tracking table. */
+        while ((ret = cursor->next(cursor.get())) == 0) {
+            testutil_check(cursor->get_key(cursor.get(), &tracked_ts));
+            testutil_check(cursor->get_value(cursor.get(), &tracked_op_type));
+
+            /* There are only two types of operations in this test. */
+            op_type = static_cast<tracking_operation>(tracked_op_type);
+            testutil_assert(
+              op_type == tracking_operation::INSERT || op_type == tracking_operation::CUSTOM);
+
+            if (num_records > 0) {
```

Comment:
Can you expand on what and how we're validating here? I was expecting something more like
```
int current_cache_size = 0;

while(cursor.next()) {
    if(cursor.value.op == CUSTOM) {
        current_cache_size = cursor.value.cache_size 
    } else if (cursor.value.op == INSERT) {
        assert(current_cache_size > 1MB);
    }
}
```

Reply:
We are not writing data to the tracking table when the cache size is 1MB, see [here](https://github.com/wiredtiger/wiredtiger/pull/7902/files#diff-e12c1ae126af9008c48745f0d1603ef8ad756233627ccc445aa530a801c00b01R86).

One thread changes the cache size every 10 seconds. 10 threads are performing inserts every 5 seconds. I am using these conditions to verify the records in the tracking table.
Happy to change back and track the `cache_size` and make sure we are always writing the right value. I thought we wanted to check the inserts happen at a time where it makes sense (i.e after the cache size has been set to a large enough value).

@tammybailey, when you review, please let me know if it confuses you as well.

Reply:
@ajmorton, I followed your approach in the end as the tests against the timing were not accurate in the end. See 77e79c833d9b3744179c68a36702e6bead369eb1.